android/fakedapp/src/main/java/com/solana/mobilewalletadapter/fakedapp/MainViewModel.kt

```diff
@@ -33,11 +36,57 @@ class MainViewModel : ViewModel() {
     private val _uiState = MutableStateFlow(UiState())
     val uiState = _uiState.asStateFlow()
 
-    private val mobileWalletAdapterClientSem = Semaphore(1) // allow only a single MWA connection at a time
+    private val compositeDisposable = CompositeDisposable()
 
-    suspend fun authorize(sender: StartActivityForResultSender) {
-        localAssociateAndExecute(sender) { client ->
-            doAuthorize(client)
+    private val mobileWalletAdapterClientSem =
```

Comment:
Wait, do we want to make the fakedapp use Rx? Or should we get a few different screens+ViewModels going here?

Reply:
Yeah definitely, different screens is needed in the future.
As mentioned in the description,  I won't merge this PR and it's mainly for discussion.

Reply:
Ooooh, sorry! Forgot that.